### PR TITLE
fix: the os disk size restriction is changed in azure.

### PR DESCRIPTION
**Pull request by andyliuliming**
**Reason for Change**:

**Issue Fixed**:

the os disk size restriction is changed to 2048 now in Azure.

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

File: pkg/api/vlabs/types.go [hunk 1]
```diff
@@ -413,7 +413,7 @@ type MasterProfile struct {
 	DNSPrefix                 string            `json:"dnsPrefix" validate:"required"`
 	SubjectAltNames           []string          `json:"subjectAltNames"`
 	VMSize                    string            `json:"vmSize" validate:"required"`
-	OSDiskSizeGB              int               `json:"osDiskSizeGB,omitempty" validate:"min=0,max=1023"`
+	OSDiskSizeGB              int               `json:"osDiskSizeGB,omitempty" validate:"min=0,max=1024"`
```
Comment (mboersma) [inline pkg/api/vlabs/types.go:416]: Shouldn't this also be 2048?

Comment (andyliuliming) [reply pkg/api/vlabs/types.go:416]: oh, yes.

Comment (andyliuliming) [reply pkg/api/vlabs/types.go]: fixed. nice catch.

File: pkg/api/vlabs/types.go [hunk 2]
```diff
@@ -486,7 +486,7 @@ type AgentPoolProfile struct {
 	Name                                string               `json:"name" validate:"required"`
 	Count                               int                  `json:"count" validate:"required,min=1,max=100"`
 	VMSize                              string               `json:"vmSize" validate:"required"`
-	OSDiskSizeGB                        int                  `json:"osDiskSizeGB,omitempty" validate:"min=0,max=1023"`
+	OSDiskSizeGB                        int                  `json:"osDiskSizeGB,omitempty" validate:"min=0,max=2048"`
```
Comment (jackfrancis) [inline pkg/api/vlabs/types.go:489]: Should we update the `DiskSizesGB` property spec as well?

Comment (andyliuliming) [reply pkg/api/vlabs/types.go:489]: from this document:
looks like it's 32T now.

will change it.

Comment (mboersma) [reply pkg/api/vlabs/types.go:489]: >  Each data disk has a maximum capacity of 32,767 gibibytes (GiB).

**Pull request merged**